src/utils.js

```diff
@@ -76,29 +97,34 @@ export const indexOf = (array, value) => {
  */
 export const createReplacer = (replacer, circularReplacer) => {
   let cache = [];
+  let initd = 0;
 
+  const callable = typeof replacer === 'function';
   const getCircular = circularReplacer || getCircularValue;
 
   return function(key, value) {
-    if (cache.length) {
-      const locationOfThis = indexOf(cache, this);
+    let replacedValue = value;
+    if (initd) { // instead of read property of object every time
```

Comment:
the reason for checking the property is because the cache length can both increase and decrease. if the cache length hits `0` again, we do not want to run this block. also, unless you use IE8 as your primary browser, the `length` property retrieval should be incredibly inexpensive.